web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -0,0 +1,180 @@
+import { Flex, ButtonIcon, Box, P2, Input } from 'design';
+import { ArrowDown, ArrowUp, Cross } from 'design/Icon';
+import { SearchAddon } from '@xterm/addon-search';
+
+export interface TerminalWithSearch {
+  getSearchAddon({
+    onDidChangeResults,
+  }: {
+    onDidChangeResults: ({
+      resultIndex,
+      resultCount,
+    }: {
+      resultIndex: number;
+      resultCount: number;
+    }) => void;
+  }): SearchAddon;
+}
+
+export const TerminalSearch = ({
+  xTerm,
+  show,
+  onClose,
+  onOpen,
+  focusTerm,
+  isSearchKeyboardEvent,
+}: {
+  xTerm: TerminalWithSearch;
+  show: boolean;
+  focusTerm(): void;
+  onClose(): void;
+  onOpen(): void;
+  isSearchKeyboardEvent: (e: KeyboardEvent) => boolean;
+}) => {
+  const theme = useTheme();
+  const searchInputRef = useRef<HTMLInputElement>();
+  const [searchValue, setSearchValue] = useState('');
+  const [searchResults, setSearchResults] = useState<{
+    resultIndex: number;
+    resultCount: number;
+  }>({ resultIndex: 0, resultCount: 0 });
+
+  const searchAddon = xTerm?.getSearchAddon({
+    onDidChangeResults: setSearchResults,
+  });
+
+  const search = (value: string, direction: 'next' | 'previous') => {
+    const match = theme.colors.terminal.brightYellow;
+    const activeMatch = theme.colors.terminal.yellow;
+    setSearchValue(value);
+    const opts = {
+      regex: true,
+      caseSensitive: false,
+      decorations: {
+        matchOverviewRuler: match,
+        activeMatchColorOverviewRuler: activeMatch,
+        matchBackground: match,
+        activeMatchBackground: activeMatch,
+      },
+    };
+
+    if (direction === 'next') {
+      searchAddon.findNext(value, opts);
+    } else {
+      searchAddon.findPrevious(value, opts);
+    }
+  };
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    search(e.target.value, 'next');
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      search(e.currentTarget.value, 'next');
+    }
+  };
+
+  const searchNext = () => {
+    search(searchValue, 'next');
+  };
+
+  const searchPrevious = () => {
+    search(searchValue, 'previous');
+  };
+
+  const onEscape = useCallback(() => {
+    onClose();
+    searchAddon.clearDecorations();
+    focusTerm();
+  }, [onClose, searchAddon, focusTerm]);
+
+  useEffect(() => {
+    const listener = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        onEscape();
+        e.preventDefault();
+      }
+      if (isSearchKeyboardEvent(e)) {
+        onOpen();
+        searchInputRef.current?.focus();
+        e.preventDefault();
+      }
+    };
+
+    document.addEventListener('keydown', listener);
+
+    return () => {
+      document.removeEventListener('keydown', listener);
+    };
+  }, [onEscape, onOpen, searchInputRef, xTerm, isSearchKeyboardEvent]);
+
+  if (!show) {
+    return;
+  }
+
+  return (
+    <SearchInputContainer alignItems="center" gap={3}>
```

Comment:
It's not a big thing but styling of the search and file transfer is a bit inconsistent. The closing icons and margins are different, one container has a shadow and the other does not:
<img width="536" alt="image" src="https://github.com/user-attachments/assets/180f16f0-68a8-446b-a811-36899d88fcad">

Reply:
I can update the styles. I worked on the styles with the design team, but there wasn't an official "design", so I can keep the intention behind the shape of it all, but make it more consistent with the file transfers box. thanks for the great attentional to detail!